Issue138

Taktar opened the pull request:
Fixes #138

What this adds

Script 1 — extract_rotations.py
Parses a slide with the side-by-side pelvis images, reads <a:xfrm rot> for each image, converts EMUs→degrees (÷ 60,000), and writes data_extraction/annotations/rotations.json:

- bone_set, display_format
- left_image_rotation & right_image_rotation (both EMU and degree)
- extracted_from_slide, applies_to_all_slides_with_format, verified_slides

Script 2 — extract_lines_labels.py
For a given slide, finds white text labels and white straight connectors/arrows, resolves text hyperlinks via the slide’s .rels, associates each label with up to two nearest lines, and writes data_extraction/annotations/slide{n}_text_annotations.json:

- text_box (x, y, width, height, rotation, shape_id)
- has_hyperlink + { target_slide, rId } when present
- pointer_lines (start/end coords, width, arrow head, shape_id)
- target_regions (line endpoints)
- total_text_annotations

How to run
**Rotations (once, from any side-by-side slide)**
python3 data_extraction/scripts/extract_rotations.py \
  --slides-dir data_extraction/fixtures/slides \
  --slide 2 \
  --out data_extraction/annotations/rotations.json

**Text + lines per slide**
python3 data_extraction/scripts/extract_lines_labels.py \
  --slides-dir data_extraction/fixtures/slides \
  --rels-dir   data_extraction/fixtures/rels \
  --slide 2 --out data_extraction/annotations/slide2_text_annotations.json

python3 data_extraction/scripts/extract_lines_labels.py \
  --slides-dir data_extraction/fixtures/slides \
  --rels-dir   data_extraction/fixtures/rels \
  --slide 3 --out data_extraction/annotations/slide3_text_annotations.json

**Outputs produced**
data_extraction/annotations/
  rotations.json
  slide2_text_annotations.json
  slide3_text_annotations.json

Quick verification I ran (and passed):
- Rotation sanity — left ≈ −16.999°, right 0.0° from EMUs:
- Printed values and assertions on tolerance ✅
- Annotation counts — total_text_annotations matches array length for each slide ✅
- Geometry — every label has 1–2 pointer_lines; target_regions have non-negative coordinates ✅
- Hyperlinks — resolved to correct target_slide via .rels (e.g., slide2: Ilium→3; Pubis→15; Ischium→9; slide3 targets 5/6/7/8 as expected) ✅

All of the above follow the instructions precisely:
- EMU→degree conversion, negative = CCW, positive = CW.
- Left vs right image determined by lower vs higher x (a:off x).
- Only white text and white lines are considered.
- Each text can have up to two pointer lines; hyperlinks (when present) are captured.

Notes:
- The rotation pair in rotations.json is marked as applying to all slides using the same side-by-side format; we verified on another slide and recorded in verified_slides.
- JSON structure matches the shapes/lines in the slide XMLs (shape IDs included for traceability).

UcheWendy commented on data_extraction/annotations/slide2_text_annotations.json:
```diff
@@ -0,0 +1,346 @@
+    "padding_emu": 4000.0,
+    "snap_emu": 8000.0
+  }
+}
```
shows five labels in slide 2, which is accurate
Text box location and size 
pointer lines that show  connecting lines from the label to the bone
Target regions

looks really good. this will help in allowing you replicate this/ overlay annotations on images next sprint.

Good work

UcheWendy commented on data_extraction/scripts/bony_pelvis_rotation.py:
```diff
@@ -0,0 +1,211 @@
+    print(f"metadata -> {args.out_metadata}")
+
+if __name__ == "__main__":
+    main()
```
great work on the script. results gotten from scripts are accurate

UcheWendy commented on data_extraction/annotations/template_bony_pelvis.json:
```diff
@@ -0,0 +1,29 @@
+    },
+    "norm_basis": "two-image-union"
+  }
+}
```
great work on this. you succeeded in creating a flexible layout template that lets you recreate the exact side-by-side bone image arrangement at any size - perfect for responsive web design like our app or different display resolutions.

UcheWendy commented on data_extraction/scripts/bony_pelvis_text_labels.py:
```diff
@@ -0,0 +1,173 @@
+    print(f"Wrote {out_path}")
+
+if __name__ == "__main__":
+    main()
```
acknowledged... great work... looking forward to the complete work to get all text labels

UcheWendy reviewed (commented):
very impressive work... look forward to the complete work

Taktar commented:
@UcheWendy 

Wrapped up the remaining slides. Added slide 4–20 XML + .rels, regenerated the two-image template (rep: slide 2), and updated bony_pelvis_metadata.json with audit at 5% (all passing). Generated per-slide text annotation JSONs for slides 4–20; slides 8, 18–20 are single-image so they’re excluded from the template/audit but have their own slide*_text_annotations.json. Also sorted target_regions by distance from label centers as discussed.

Everything looks consistent with slides 2–3—ready to merge. 

just to clarify, before the merge, you said the rels and xmls files should be deleted so there is no duplicate later right?
